backend/models.py

```
@@ -270,22 +297,17 @@ class Hint(db.Model):
 
-    # Check constraint ensuring that the associated module is of type 'CHALLENGE'
-    def __init__(self, text, order, module_id):
-        self.text = text
-        self.order = order
-        self.module_id = module_id
-        self.validate_module_type()
-
-    def validate_module_type(self):
-        module = Module.query.get(self.module_id)
+    @validates("module_id")
+    def validate_module_type(self, key, module_id):
```

Ruff (0.8.2)

Missing return type annotation for public function `validate_module_type`

(ANN201)